Tasks/ChefKnifeV1/ChefKnife.ps1

```diff
@@ -31,6 +31,13 @@ try
     #setting up chef repo with the chef subscription details fetched before 
     Initialize-ChefRepo $connectedServiceDetails 
     Invoke-Expression -Command $scriptCommand
+    $featureFlags = @{
+        failDeprecatedBuildTask  = [System.Convert]::ToBoolean($env:FAIL_DEPRECATED_BUILD_TASK)
+    }
+	if ($featureFlags.failDeprecatedBuildTask)
+	{
+		throw "The ChefKnife@1 (Run scripts with Knife commands on your Chef workstation) task has been deprecated since March 5, 2018 and will soon be retired. To continue to use Chef Knife, use the Knife client directly from a bash/pwsh/script task. See https://docs.chef.io/workstation/knife_client/."
```

Comment:
Shouldn't this string be sourced from task.json?

Reply:
For PowerShell based tasks, in order to read the strings from resource file generated from task.json, we need to add another module (VstsTaskSdk) in the dependencies of the task. For a deprecated Task, I believe we shouldn't add a new dependency.